### ISSUE TITLE
A question about formatting_prompts_func

### ISSUE THREAD

**Issue by sunzx8** [mzbac/llama2-fine-tune#1]
Hi, I notice that in your code in make :

 def formatting_prompts_func(example):
 output_texts = []
 
 for i in range(len(example['prompt'])):
 
 text = f"### Input: ```{example['prompt'][i]}```\n ### Output: {example['completion'][i]}"
 
 output_texts.append(text)
 
 return output_texts

But i find that you did not split the Input and Output . How can the model judge which part is the output?
It is more like a way to input a whole text which contains both input and output
I tried your way and make my function:
def formatting_prompts_func(example):
    a_list=[]
    for  i in range(len(example['input'])):
        text = (
            f"### Instruction: {example['instruction'][i]}\n" +
            f"### Input: {example['input'][i]}\n" +
            f"### Output: {example['output'][i]}"
        )
        print(text)
        a_list.append(text)
    return a_list
    
    But after one round training the loss becomes 0
<img width="532" alt="image" src="https://github.com/mzbac/llama2-fine-tune/assets/136688632/06efa592-1a6f-42e4-bf4d-b9028232f4a4">

Iam not sure what;s going on , It's that because i don't specify the output?

**Comment by mzbac** [comment]
GPT models are text completion models. You don't have to make the training data as pairs, see details in trl's sft trainer source code. https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L57

**Comment by sunzx8** [comment]
> GPT models are text completion models. You don't have to make the training data as pairs, see details in trl's sft trainer source code. https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L57

Hi, thank you for your answer, I know that it should not be pairs, I mean why in sfttrainer we do not add labels,EOS token ID to mask tokens that do not require predicting

**Comment by mzbac** [comment]
From my understanding, the GPT model is a text generation model. You would use the seq2seq trainer to mask out the input and train the model only preditct the output, like Qlora did. I am not exactly sure why SFT trainer prefers ConstantLengthDataset, but I have tried both methods and didn't notice much difference in results.

**Comment by sunzx8** [comment]
> From my understanding, the GPT model is a text generation model. You would use the seq2seq trainer to mask out the input and train the model only preditct the output, like Qlora did. I am not exactly sure why SFT trainer prefers ConstantLengthDataset, but I have tried both methods and didn't notice much difference in results.
I see, thank you!
